ngboost/distns/poisson.py

```diff
@@ -0,0 +1,67 @@
+from ngboost.distns import RegressionDistn
+from ngboost.scores import LogScore
+import scipy as sp
+import numpy as np
+from scipy.stats import poisson as dist
+from scipy.special import factorial
+from scipy.optimize import minimize
+
+### Helpers ####
+def negative_log_likelihood(params, data):
+    return -dist.logpmf(np.array(data), params[0]).sum()
+
+def isinteger(x):
+    return np.equal(np.mod(x, 1), 0)
+
+
+## NGBoost Classes ##
+class PoissonLogScore(LogScore):
+    def score(self, Y):
+        return -self.dist.logpmf(Y)
+    
+    def d_score(self, Y):
+        D = np.zeros((len(Y), 1))
+        D[:, 0] = self.mu - Y
+        return D
+    
+class Poisson(RegressionDistn):
+
+    n_params = 1
+    scores = [PoissonLogScore] 
+
+    def __init__(self, params):
+        # save the parameters
+        self._params = params
+        
+        # create other objects that will be useful later
+        self.logmu = params[0]
+        self.mu = np.exp(self.logmu)
+        self.dist = dist(mu=self.mu)
+
+    def fit(Y):
+        assert(np.all([np.equal(np.mod(y, 1), 0) for y in Y])), "All Poisson target data must be discrete integers"
```

Comment:
Not sure if I am missing something with the input `Y` and potential shapes, but if `Y` is a numpy array we can run this faster as a fully vectorised calc without the list comprehension.  Even if `Y` is allowed to be a list or other np.array-able object there is a decent-sized speed boost available.

```suggestion
        assert(np.equal(np.mod(y, 1), 0).all), "All Poisson target data must be discrete integers"
```